tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,89 @@
+``` 
+where `self.train_set.item_text` is set to our instantiated text modality `item_text_modality`, `n_items` is number of training items, and the `batch_bow()` function returns the bag-of-word vectors of the specified item ids, in our case we want the text features for all training items: `np.arange(n_items)`.
+
+**Aligning auxiliary information with the set of training items.** Another important aspect worth mentioning at this level is that, we don't have to take extra actions to align the set of training movies with their plots. They are already aligned. That is, the first row in `text_feature` corresponds to training `movie_id: 1`, the second row to training `movie_id: 2`, and so on. This is made possible thanks to passing `item_text_modality` through the evaluation (splitting) method as we shall see shortly. 
+
+### Evaluation
+
+We are now left with the standard steps to fit and evaluate a recommender model using Cornac. We use train-test splitting as our evaluation method:
+```Python
+from cornac.eval_methods import RatioSplit
+
+ratio_split = RatioSplit(data=rating_data, test_size=0.2, exclude_unknowns=True,
+                         item_text=item_text_modality, verbose=True,
+                         seed=123, rating_threshold=0.5)
+``` 
+The text modality passed for the evaluation method. As mentioned earlier, this make it possible to avoid the tedious process of aligning the set of training items with their auxiliary data. Next, we need to instantiate the CLD model and evaluation metrics,
+```Python
+cdl = cornac.models.CDL(k=20, autoencoder_structure=[50], max_iter=100,
+                        learning_rate=0.01, lambda_u=0.01, lambda_v=0.01,
+                        lambda_w=0.01, lambda_n=1000, vocab_size=5000)
+                        
+rec_100 = cornac.metrics.Recall(k=100)
+``` 
+Finally, we put everything together and run our experiment.
+```Python
+exp = cornac.Experiment(eval_method=ratio_split,
+                        models=[cdl],
+                        metrics=[rec_100])
+exp.run()
+```
+Output:
+```
+    | Recall@100 | Train (s) | Test (s)
+--- + ---------- + --------- + --------
+CDL |     0.5494 |   42.1279 |   0.3018
+```
+Note that one may achieve higher results by careful parameter tuning. The purpose here is to illustrate how to handle auxiliary data using Cornac.  
+
+## Other Modality classes
+
+The usage of the GraphModality and ImageModality, to deal with graph and visual auxiliary data, follows the same principles as above. The following two examples involve the GraphModality class to handle item graph (network) auxiliary data: [c2pf_example](../examples/c2pf_example.py), [mcf_example](../examples/mcf_office.py). For a usage example of the ImageModality one may refer to [vbpr_tradesy](../examples/vbpr_tradesy.py). The `cornac.data` module's [documentation](https://cornac.readthedocs.io/en/latest/data.html) is also a good resource to know more about the modality classes. 
```

Comment:
The usage of the `GraphModality` and `ImageModality` to deal with graph and visual auxiliary data follows the same principles as above. The following two examples, [c2pf_example](../examples/c2pf_example.py), [mcf_example](../examples/mcf_office.py), involve `GraphModality` to handle item network. For the `ImageModality`, one may refer to [vbpr_tradesy](../examples/vbpr_tradesy.py) example. The `cornac.data` module's [documentation](https://cornac.readthedocs.io/en/latest/data.html) is also a good resource to know more about the modality classes.

Reply:
I rephrased this part according to your comments.